Classes/Code/ScriptTagBuilder.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Code;
+
+use Brotkrueml\MatomoIntegration\Event\EnrichScriptTagEvent;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+final class ScriptTagBuilder
+{
+    private ServerRequestInterface $request;
+    private EventDispatcherInterface $eventDispatcher;
+
+    private string $tagName = 'script';
+    private string $code = '';
+
+    /**
+     * @var array<string>
+     */
+    private array $attributes = [];
+
+    public function __construct(
+        string $code,
+        ?ServerRequestInterface $request = null,
+        ?EventDispatcherInterface $eventDispatcher = null
+    ) {
+        $this->code = $code;
+        $this->request = $request ?? $this->getRequest();
+        $this->eventDispatcher = $eventDispatcher ?? $this->getEventDispatcher();
+    }
+
+    /**
+     * Build the script start script tag with attributes
+     */
+    public function build(): string
+    {
+        $this->collectAttributes();
+
+        $prepend = '';
+        if ($this->attributes !== []) {
+            $prepend = ' ';
+        }
+
+        return '<' . $this->tagName . $prepend . implode(' ', $this->attributes) . '>' . $this->code . '</' . $this->tagName . '>';
+    }
+
+    /**
+     * Collect all attributes
+     */
+    private function collectAttributes(): void
+    {
+        /** @var EnrichScriptTagEvent $enrichScriptTagEvent */
+        $enrichScriptTagEvent = GeneralUtility::makeInstance(EnrichScriptTagEvent::class, $this->request);
+        $enrichScriptTagEvent = $this->eventDispatcher->dispatch($enrichScriptTagEvent);
+        if (! ($enrichScriptTagEvent instanceof EnrichScriptTagEvent)) {
+            return;
+        }
+
+        if ($enrichScriptTagEvent->getId() !== '') {
```

Comment:
What if we just collect the attributes as key/values:

    if ($event->getId() !== '') {
        $attributes['id'] = $event->getId();
    }
    if ($event->getType() !== '') {
        $attributes['type'] = $event->getId();
    }
    foreach ($event->getDataAttributes() as $name => $value) {
        $attributes['data-' . $name] = $value;
    }

The we can use something like this:
https://3v4l.org/mkTr4

I don't like very much the approch to build the attributes string in the data structure. We should collect the attributes first and then format them accordingly. I think it's a cleaner approach. What's your opinion?

This way we can get rid of the `addAttributeWithValue()` and `addAttribute()` methods.